src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,36 @@
+package duke.tasks;
+
+import java.time.LocalDate;
+
+/**
+ * Represents a Deadline task that is created upon user input.
+ */
+public class Deadline extends Task {
+    protected String by;
+
+    /**
+     * Constructor for a Deadline Task which stores the description of
+     * the task and the deadline of it.
+     * @param description Task description.
+     * @param by Task deadline.
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+        this.date = LocalDate.parse(this.by);
```

Comment:
The utility of by and date seems to be overlapping here, maybe you the usage of String by can be completely avoided.

Reply:
Agreed